Cython/Compiler/ExprNodes.py

```diff
@@ -2055,6 +2057,8 @@ def analyse_target_declaration(self, env):
         if self.entry.as_module:
             # cimported modules namespace can shadow actual variables
             self.entry.is_variable = 1
+        if not self.entry.annotation and self.annotation:
+            self.entry.annotation = self.annotation.expr
```

Comment:
Shouldn't this just be `self.annotation` on the right hand side?

Reply:
I don't think this is the correct behavior either - afaict CPython doesn't support annotating a variable after it's been declared.

Reply:
True. Why was this change necessary then?

Reply:
It wasn't - I forgot I'd added that in while debugging why entries were lacking annotations when I thought they should have them. I've removed it 👍